test/e2e/nodeportlocal_test.go

```diff
@@ -0,0 +1,389 @@
+// +build !windows
+
+// Copyright 2021 Antrea Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package e2e
+
+import (
+	"encoding/json"
+	"fmt"
+	"strings"
+	"testing"
+	"time"
+
+	"github.com/stretchr/testify/require"
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/util/wait"
+
+	"github.com/vmware-tanzu/antrea/pkg/agent/nodeportlocal/k8s"
+)
+
+const (
+	defaultStartPort = 40000
+	defaultEndPort   = 41000
+	updatedStartPort = 42000
+	updatedEndPort   = 43000
+	nplSvcAnnotation = "nodeportlocal.antrea.io/enabled"
+)
+
+func getNPLAnnotation(t *testing.T, data *TestData, r *require.Assertions, testPodName string) (string, *PodIPs) {
+	var nplAnn string
+	var testPodIP *PodIPs
+	var found bool
+	var err error
+	err = wait.Poll(time.Second, defaultTimeout, func() (bool, error) {
```

Comment:
this seems sub-optimal: `podWaitFor` starts a `Poll`, which implies a 1s sleep. Then we call `podWaitForIPs`, which in turns call `podWaitFor`, with another `Poll` and another 1s sleep.

I think you should replace this "top-level" `wait.Poll` call by a single call to `podWaitFor`, with the `PodCondition` parameter that works for your use case.

Reply:
Done